jupyter_compare_view/compare.py

```diff
@@ -1,48 +1,122 @@
+import base64
+import enum
+import io
 import json
-import os
+import typing
 import uuid
 from pathlib import Path
 from jinja2 import Template, StrictUndefined
-from IPython.core.display import HTML, JSON
-from IPython.display import display
+import IPython
+import PIL
+
+
+ImageLike = typing.TypeVar('ImageLike')
+
+
+def img2bytes(img: ImageLike, format: str = 'jpeg') -> bytes:
+    im_file = io.BytesIO()
+    if isinstance(img, PIL.Image.Image):
+        img.save(im_file, format=format)
+    else:
+        # anything other that can be displayed with plt.imshow
+        import matplotlib.pyplot as plt
+        
+        plt.imsave(im_file, img, format=format)
```

Comment:
```suggestion
        plt.imsave(im_file, img, format=format, cmap= "gray")
```
otherwise, gray images would look like this
<img width="413" alt="image" src="https://user-images.githubusercontent.com/44469195/205435841-c549bff3-e43c-4c5a-93d0-3066b95f2b8f.png">

Reply:
I think it's better to respect `rcParams['image.cmap']` so I added `cmap` argument to `compare`.